docs/applications/junebug_message_api.rst

```diff
@@ -0,0 +1,247 @@
+
+.. py:data:: transport_type
+   :type: str
+
+   The transport_type to use for non-reply outbound messages. Defaults to ``sms``.
+
+.. py:data:: default_from_addr
+   :type: str
+
+   The from address to be used for non-reply outbound messages that don't specify a from address. By default, non-reply outbound messages are required to specify a from address.
+
+.. py:data:: allow_expired_replies
+   :type: bool
+
+   If ``True``, outbound messages with both ``to`` and ``reply_to`` set will be sent as non-reply messages if the ``reply_to`` message can't be found. Defaults to ``False``.
+
+.. py:data:: state_cache_class
+   :type: str
+
+   The python path to the class used for the application state cache. This class is resposible for caching inbound messages for outbound replies and event info for outbound messages. Defaults to ``vumi2.applications.junebug_message_api.junebug_state_cache.InMemorySmppCache``, which stores the data in memory. See :ref:`junebug-state-caches` for a list of state caches.
+
+.. py:data:: state_cache_config
+   :type: dict
+
+   The config that :py:data:`state_cache_class` requires. See :ref:`junebug-state-caches` for details.
+
+.. py:data:: request_timeout
+   :type: float
+
+   The maximum time allowed (in seconds) for outbound message request handling. Defaults to 240.
+
+.. py:data:: mo_message_url_timeout
+   :type: float
+
+   Maximum time allowed (in seconds) for inbound message HTTP requests. Defaults to 10.
+
+.. py:data:: event_url_timeout
+   :type: float
+
+   Maximum time allowed (in seconds) for event HTTP requests. Defaults to 10.
+
+How it works
+^^^^^^^^^^^^
+
+The application worker listens on HTTP for outbound messages from the external application and forwards them over AMQP to a router or transport. Inbound messages and events are forwarded to the external application over HTTP.
+
+Outbound message API
+""""""""""""""""""""
+
+.. http:post:: <base_url_path>/messages
+
+   Send an outbound (mobile terminated) message.
+
+   :<json str to: The address (e.g. MSISDN) to send the message to. If :py:data:`allow_expired_replies` is set, the ``to`` parameter is used as a fallback in case the value of the ``reply_to`` parameter does not resolve to an inbound message.
+
+   :<json str from: The address the message is from. May be ``null`` if :py:data:`default_from_addr` is configured.
+
+   :<json str group: If supported by the transport or router, the group to send the messages to. Not required, and may be ``null``.
+
+   :<json str reply_to: The uuid of the message being replied to if this is a response to a previous message. Important for session-based transports like USSD. Optional.
+      If :py:data:`allow_expired_replies` is set, ``to`` and ``from`` will be used as a fallback in case ``reply_to`` does not resolve to an inbound message.
+      The default settings allow 10 minutes to reply to a message, after which an error will be returned.
```

Comment:
Seems like we allow 24 hours by default on the in-memory cache

Reply:
Thanks. I must have forgotten to update the time when I copied the old Junebug docs over.